### PR TITLE
Add cyclical learning rate

### PR DESCRIPTION
**Pull request by Tauranis**
Add cyclical learning rate as reported by `Leslie N Smith. Cyclical learning rates for training neural networks. arXiv preprint arXiv:1506.01186v3, 2016.`

See [this example](https://postimg.cc/image/vzwtaxzhj/) to understand what a cyclical learning rate is about

The function can be easily couple to others like `tf.train.exponential`

  ```python
  ...
  cosine_lr = tf.train.cosine_cyclical(starter_learning_rate, global_step,
                                            decay_steps, end_learning_rate, T)
  cosine_exp_lr = tf.train.exponential_decay(cosine_lr,global_step,300,0.7,staircase=False)

  ```

  For the a better comprehension, see [this example](https://postimg.cc/image/8tcgv2xjb/)  of how the snippet above looks like in practice.

### CODE REVIEW

**Comment by Tauranis** [comment]
rodrigofp-cit add on CLA

**Comment by rodrigofp-cit** [comment]
Add rodrigofp-cit on PR

**Comment by Tauranis** [comment]
rodrigofp-cit Added
